dstep/translator/MacroDefinition.d

```diff
@@ -0,0 +1,1859 @@
+        }
+        else if (acceptPunctuation!("(")(local))
+        {
+            if (acceptPunctuation!(")")(local))
+            {
+                CallExpr subexpr = new CallExpr;
+                subexpr.expr = expr;
+                subexpr.args = [];
+                expr = subexpr;
+            }
+            else
+            {
+                auto args = parseArgsList(local, table);
+
+                if (args is null)
+                    break;
+
+                if (!acceptPunctuation!(")")(local))
+                    break;
+
+                CallExpr subexpr = new CallExpr;
+                subexpr.expr = expr;
+                subexpr.args = args;
+                expr = subexpr;
+            }
+        }
+        else if (acceptPunctuation!(".")(local) && acceptIdentifier(local, spelling))
+        {
+            DotExpr subexpr = new DotExpr;
+            subexpr.subexpr = expr;
+            subexpr.identifier = spelling;
+            expr = subexpr;
+        }
+        else if (acceptPunctuation!("->")(local) && acceptIdentifier(local, spelling))
+        {
+            ArrowExpr subexpr = new ArrowExpr;
+            subexpr.subexpr = expr;
+            subexpr.identifier = spelling;
+            expr = subexpr;
+        }
+        else if (acceptPunctuation!("++", "--")(local, spelling))
+        {
+            UnaryExpr subexpr = new UnaryExpr;
+            subexpr.subexpr = expr;
+            subexpr.operator = spelling;
+            subexpr.postfix = true;
+            expr = subexpr;
+        }
+        else
+        {
+            break;
+        }
+    }
+
+    tokens = local;
+
+    return expr;
+}
+
+Expression parseUnaryExpr(ref TokenRange tokens, Cursor[string] table)
+{
+    auto local = tokens;
+
+    string spelling;
+
+    if (accept!("++", "--")(local, spelling, TokenKind.punctuation))
+    {
+        Expression subexpr = parseUnaryExpr(local, table);
+
+        if (subexpr !is null)
+        {
+            UnaryExpr expr = new UnaryExpr;
+            expr.subexpr = subexpr;
+            expr.operator = spelling;
+            tokens = local;
+            return expr;
+        }
+    }
+
+    if (accept!("&", "*", "+", "-", "~", "!")(local, spelling, TokenKind.punctuation))
+    {
+        Expression subexpr = parseCastExpr(local, table);
+
+        if (subexpr !is null)
+        {
+            UnaryExpr expr = new UnaryExpr;
+            expr.subexpr = subexpr;
+            expr.operator = spelling;
+            tokens = local;
+            return expr;
+        }
+    }
+
+    if (accept!("sizeof")(local, spelling, TokenKind.keyword))
+    {
+        Expression subexpr = parseUnaryExpr(local, table);
+
+        if (subexpr !is null)
+        {
+            UnaryExpr expr = new UnaryExpr;
+            expr.subexpr = subexpr;
+            expr.operator = spelling;
+            tokens = local;
+            return expr;
+        }
+
+        // FIXME: unary-expression ::= sizeof ( type-name )
```

Comment:
Parsing of C-style sizeof is type dependent too (you need to know the names of types). I added that FIXME when translation of types was not fully implemented. But as it is handled quite good now, I've just added a commit that implements parsing and translation of type-dependent `sizeof` operators.